### DISCUSSION TITLE
Lesson 4 - Bracket issue

### DISCUSSION THREAD

**Discussion by EdenSwack** [smartcontractkit/full-blockchain-solidity-course-py#1747] [Q&A] [answered]
`
transacition=  SimpleStorage.constructor().buildTransaction (
    {
        "chainId:" chainId,
        "from:" my_address,
        "nonce:" nonce,
    }
)   `

I get errors:
"(" was not closed
"{" was not closed
<img width="645" alt="Screen Shot 2022-10-15 at 18 59 05" src="https://user-images.githubusercontent.com/93931254/195996140-888f659d-36ef-474a-a4fb-464100c4a793.png">

For bolded brackets

**Comment by cromewar** [comment]
Hello @EdenSwack 
You have syntax error on your code, each `(` should have its own `)` and all `{` also needs its own `}`, please check out the course repo and try to match the exact same syntax in order to solve this, cheers.

**Reply by eswack** [reply]
Can you say which line? I don't understand where the brackets are not closed.
This is from the repo: 

```
transaction = SimpleStorage.constructor().buildTransaction(
    {
        "chainId": chain_id,
        "gasPrice": w3.eth.gas_price,
        "from": my_address,
        "nonce": nonce,
    }
)
```

and this is mine: 
```

transacition=  SimpleStorage.constructor().buildTransaction(
    {
        "chainId:" chainId,
        "from:" my_address,
        "nonce:" nonce,
    }
)   
```

**Comment by Niferu** [comment, marked as answer]
Hello @eswack, @EdenSwack 
You got this syntax error because you did not use `:` in proper place as those should be placed after `""`, so in your case it looks like `"chainId:"` and it should be `"chainId":` I'm glad I could help :)

**Reply by EdenSwack** [reply]
Damn how did I not notice. Thanks!
